### ISSUE TITLE
Keras Model with Reshape + LSTM Fails When Imported

### ISSUE THREAD

**Issue by kullanici0606** [deeplearning4j/deeplearning4j#4556]
#### Importing Keras model Reshape + LSTM fails with dimensionality problems.
I have a keras 2 model (using keras function API, as recommended in #4552 ) where there are Reshape layers before LSTM layers, but when I import the model, this fails with following exception:

`java.lang.IllegalStateException: Mis matched lengths: [2560] != [256] - Array 1 shape: [10, 256], array 2 shape: [1, 256]
	at org.nd4j.linalg.util.LinAlgExceptions.assertSameLength(LinAlgExceptions.java:41)
	at org.nd4j.linalg.api.ops.BaseTransformOp.<init>(BaseTransformOp.java:174)
	at org.nd4j.linalg.api.ops.impl.transforms.Set.<init>(Set.java:41)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.assign(BaseNDArray.java:1274)
	at org.deeplearning4j.nn.layers.BaseLayer.setParam(BaseLayer.java:204)
	at org.deeplearning4j.nn.layers.wrapper.BaseWrapperLayer.setParam(BaseWrapperLayer.java:271)
	at org.deeplearning4j.nn.modelimport.keras.KerasLayer.copyWeightsToLayer(KerasLayer.java:295)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelUtils.copyWeightsToModel(KerasModelUtils.java:76)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.getComputationGraph(KerasModel.java:377)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.getComputationGraph(KerasModel.java:363)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:107)`

My model code (Also attaching py file and model itself: [reshape.zip](https://github.com/deeplearning4j/deeplearning4j/files/1667779/reshape.zip))
`input = Input(shape=(10,))`
`   reshape_model = Reshape(target_shape=(10, 1))(input)`
`    reshape_model = LSTM(64)(reshape_model)`
`    reshape_model = Activation('relu')(reshape_model)`
`    reshape_model = Dropout(0.5)(reshape_model)`
`    reshape_model = Dense(1)(reshape_model)`
`    reshape_model = Activation('sigmoid')(reshape_model)`

`    model = Model(inputs=[input], outputs=reshape_model)`
`    model.compile(loss='binary_crossentropy', optimizer='Adam',  metrics=['accuracy'])`

Import code:
`ComputationGraph graph = KerasModelImport.importKerasModelAndWeights(reshape_model.h5", false);`

#### Expected behavior
Should import the model

#### Encountered behavior
Fails with java.lang.IllegalStateException: Mis matched lengths: [2560] != [256] - Array 1 shape: [10, 256], array 2 shape: [1, 256]

#### Version Information
0.9.2-SNAPSHOT

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 0.9.2-SNAPSHOT
* Keras version: 2.1.1
* Tensorflow version 1.4.0
* Windows 10
* No CUDA
* No NVIDIA driver

**Comment by kullanici0606** [comment]
Hi @maxpumperla ,

I tried to import the model with the updated code and it seems to import without any problems. However when I try to make a prediction, it fails with the exception below. am I doing something wrong or is this another problem not related to reshape model import?

Thanks.

```java
double [] inputArr = new double[10];
for(int i=0; i < inputArr.length; i++) {
	inputArr[i] = random.nextInt();
}
		
INDArray input = Nd4j.create(inputArr).reshape(10);
		
KerasModelBuilder builder = new KerasModel().modelBuilder().modelHdf5Filename(modelPath)
				.enforceTrainingConfig(false);
		
KerasModel model = builder.buildModel();
INDArray result = model.getComputationGraph().outputSingle(input);
```
Exception:
```java
Exception in thread "main" org.deeplearning4j.exception.DL4JInvalidInputException: Received input with size(1) = 1 (input array shape = [10, 1]); input.size(1) must match layer nIn size (nIn = 10)
	at org.deeplearning4j.nn.layers.recurrent.LSTMHelpers.activateHelper(LSTMHelpers.java:164)
	at org.deeplearning4j.nn.layers.recurrent.LSTM.activateHelper(LSTM.java:184)
	at org.deeplearning4j.nn.layers.recurrent.LSTM.activate(LSTM.java:158)
	at org.deeplearning4j.nn.layers.recurrent.LastTimeStepLayer.activate(LastTimeStepLayer.java:88)
	at org.deeplearning4j.nn.graph.vertex.impl.LayerVertex.doForward(LayerVertex.java:105)
	at org.deeplearning4j.nn.graph.ComputationGraph.feedForward(ComputationGraph.java:1680)
	at org.deeplearning4j.nn.graph.ComputationGraph.feedForward(ComputationGraph.java:1566)
	at org.deeplearning4j.nn.graph.ComputationGraph.silentOutput(ComputationGraph.java:1835)
	at org.deeplearning4j.nn.graph.ComputationGraph.output(ComputationGraph.java:1895)
	at org.deeplearning4j.nn.graph.ComputationGraph.outputSingle(ComputationGraph.java:1868)
	at org.deeplearning4j.nn.graph.ComputationGraph.outputSingle(ComputationGraph.java:1854)
	at org.deeplearning4j.nn.graph.ComputationGraph.outputSingle(ComputationGraph.java:1772)
```

**Comment by maxpumperla** [comment]
Try something like

```{java}
INDArray input = Nd4j.zeros(miniBatch, 1, inputLength);
INDArray output = model.output(input);
```

**Comment by kullanici0606** [comment]
It worked, thank you very much.

**Comment by maxpumperla** [comment]
@kullanici0606 cool, thanks for the feedback. Just for documentation purposes:

As a Keras user I'd recommend not to move part of your preprocessing into the first layer. It's OK, since  Keras is flexible enough, but generally discouraged.

Part of the reason you ran into trouble here is that Keras and DL4J have different internal representations of their layer inputs. E.g. in an LSTM, the number of time steps are in position 2 in Keras and position 3 in DL4J. You could argue that `Reshape` is a little too flexible and that you shouldn't know about internals to properly define one.  Now, in mapping the frameworks that makes it a little difficult to handle without a million if else statements. :)

Great, so I'm glad this worked out.

**Comment by kullanici0606** [comment]
Thanks for the information / recommendation.
